frontend/src/views/Lobby.tsx

```diff
@@ -81,6 +85,33 @@ function LobbyPage() {
       });
   };
 
+  /**
+   * Update the difficulty setting of the room (EASY, MEDIUM, HARD, or none)
+   */
+  const updateDifficultySetting = (key: string) => {
+    if (currentUser?.nickname === host?.nickname && !loading) {
+      setLoading(true);
+      let newDifficulty: Difficulty | null = Difficulty[key as keyof typeof Difficulty];
+
+      // Set new difficulty to none if previous setting was toggled off
+      if (newDifficulty === difficulty) {
+        newDifficulty = null;
+      }
+
+      const newSettings = {
+        initiator: currentUser!,
+        difficulty: newDifficulty,
+      };
+
+      updateRoomSettings(currentRoomId, newSettings)
+        .then(() => setLoading(false))
+        .catch((err) => {
+          setLoading(false);
+          setError(err.message);
+        });
```

Comment:
Sounds good, just implemented all of this - with the one exception that I just re-used the current loading variable to track it, so it does show on the UI rather than being hidden. Lmk your thoughts on that (like if you think it should be hidden)

Reply:
Wow thanks so much for implementing all of that now! I think it's totally fine that the `loading` component is visible on the UI. Nice job!